web/api/app-profile/upload-image/index.ts

```diff
@@ -1,41 +1,33 @@
-import { getSdk as checkUserInAppDocumentSDK } from "@/legacy/api/images/graphql/checkUserInApp.generated";
-import { errorHasuraQuery, errorNotAllowed } from "@/legacy/backend/errors";
-import { getAPIServiceGraphqlClient } from "@/legacy/backend/graphql";
-import { protectInternalEndpoint } from "@/legacy/backend/utils";
+import { getSdk as checkUserInAppDocumentSDK } from "@/api/app-profile/graphql/checkUserInApp.generated";
+import { errorHasuraQuery, errorNotAllowed } from "@/api/helpers/errors";
+import { getAPIServiceGraphqlClient } from "@/api/helpers/graphql";
+import { protectInternalEndpoint } from "@/api/helpers/utils";
+import { validateRequestSchema } from "@/api/helpers/validate-request-schema";
 import { logger } from "@/lib/logger";
 import { S3Client } from "@aws-sdk/client-s3";
 import { createPresignedPost } from "@aws-sdk/s3-presigned-post";
-import { NextApiRequest, NextApiResponse } from "next";
+import { NextRequest, NextResponse } from "next/server";
+import * as yup from "yup";
 
-type RequestQueryParams = {
-  app_id: string;
-  image_type: string;
-  content_type_ending: string;
-};
+const schema = yup.object({
+  app_id: yup.string().strict().required(),
+  image_type: yup.string().strict().required(),
+  content_type_ending: yup.string().required(),
+});
 
-/**
- * Returns a signed url to upload to the predefined path in S3
- * @param req
- * @param res
- */
-
-// TODO: When we migrate this to new API, should convert to GET
-export const handleImageUpload = async (
-  req: NextApiRequest,
-  res: NextApiResponse,
-) => {
+export const POST = async (req: NextRequest) => {
   try {
-    if (!protectInternalEndpoint(req, res)) {
+    if (!protectInternalEndpoint(req)) {
       return;
     }
 
-    if (req.method !== "POST") {
-      return errorNotAllowed(req.method, res, req);
+    if (!req.method || req.method !== "POST") {
```

Comment:
@andy-t-wang I think we can leave it explicitly, but I also think that nextjs route handlers now validate method by function name (`POST` in this case)

Reply:
Can you please double-check it?

Reply:
And if it is so, then I think Nextjs will throw an error earlier, before even starting to execute endpoint code